### PR TITLE
ZUR-184 feat: ability to react to messages and images in rooms

### PR DESCRIPTION
**Pull request by Johndiddles**

### CODE REVIEW

**Comment by JulianaSau** [comment]
@tundealabi
Should this change be done in `zc_main` or `zc_messaging`?

**Comment by tundealabi** [comment]
> @tundealabi Should this change be done in `zc_main` or `zc_messaging`?

both

**Comment by JulianaSau** [comment]
Okay. Is it working on your end now? I saw some message reactions when testing other prs on local.

**Comment by tundealabi** [comment]
> Okay. Is it working on your end now? I saw some message reactions when testing other prs on localhost.

those reactions have been there

**Comment by Johndiddles** [comment]
@tundealabi I am not sure you have taken a look at it.......those reactions were not there until I worked on it.

**Comment by tundealabi** [comment]
message duplicates when adding/removing an emoji
Screenshot 2022-11-13 at 12 26 34

**Comment by JulianaSau** [comment]
@Johndiddles have you noted this?

**Comment by Johndiddles** [comment]
@JulianaSau Yes, I just did a patch on both zc_main and zc_messaging for the suggestion @tundealabi mentioned earlier.

**Comment by JulianaSau** [comment]
Add a video showing that it's not duplicating again

**Comment by Johndiddles** [comment]
@JulianaSau @tundealabi 

https://user-images.githubusercontent.com/48492384/201535791-e29c57d6-c557-401a-b973-ab76107f84b8.mp4

File: packages/ui/src/message-board/MessageBoard.jsx [hunk 1]
```diff
@@ -21,6 +21,11 @@ function MessageBoard({
   onSendAttachedFile,
   onReact
 }) {
+  console.log({ messages });
```
Comment (tundealabi) [inline packages/ui/src/message-board/MessageBoard.jsx:24]: remove the console.log. thanks

**Comment by Johndiddles** [comment]
@tundealabi console.log has been removed. thanks

**Review by tundealabi — DISMISSED** [review]
can you remove the comments? thanks

File: packages/ui/src/message-board/MessageBoard.jsx [hunk 2]
```diff
@@ -21,6 +21,10 @@ function MessageBoard({
   onSendAttachedFile,
   onReact
 }) {
+  // messages = Array.from(new Set(messages.map(a => a._id))).map(id => {
```
Comment (tundealabi) [inline packages/ui/src/message-board/MessageBoard.jsx:24]: comments here

File: packages/ui/src/message-pane/MessagePane.jsx [hunk 3]
```diff
@@ -13,11 +13,12 @@ function MessagePane({
   onEmojiClicked,
   currentUserId
 }) {
+  // console.log({ message });
```
Comment (tundealabi) [inline packages/ui/src/message-pane/MessagePane.jsx:16]: comments here

File: packages/ui/src/message-board/MessageBoard.jsx (hunk 2 above)
Comment (Johndiddles) [reply packages/ui/src/message-board/MessageBoard.jsx]: Done

File: packages/ui/src/message-pane/MessagePane.jsx (hunk 3 above)
Comment (Johndiddles) [reply packages/ui/src/message-pane/MessagePane.jsx:16]: Done

**Comment by Johndiddles** [comment]
all console.logs have been removed.....thanks

File: packages/ui/src/message-pane/MessagePane.jsx (hunk 3 above)
Comment (tundealabi) [reply packages/ui/src/message-pane/MessagePane.jsx:16]: still see a comment here

Comment (tundealabi) [reply packages/ui/src/message-pane/MessagePane.jsx:16]: // console.log({ message });

**Review by JulianaSau — CHANGES_REQUESTED** [review]
Your pr is changing the `yarn.lock` file.
Updating this file could affect other contributors so please remove it from your commits

**Comment by Johndiddles** [comment]
yarn.lock has been removed and added to .gitignore

**Comment by JulianaSau** [comment]
> yarn.lock has been removed and added to .gitignore

Don't add the lock file to `.gitignore`, just ensure you don't modify it at all. The lock file in the repo should be the same as the lock file you have on your ◀️ cal repository

**Comment by JulianaSau** [comment]
Deleting them still qualifies as modification 😅😅😅
Check your changed files in this or. You'll see that you've modified the lock files

**Comment by Johndiddles** [comment]
I didn't delete the lock files 😂. I untracked them, so it appeared in the history as deleted. now that I have added them back to the track but made sure they are the same with what is on the zc_main repo, they are appearing as additions. But I made sure they are exactly the same with what is on the repo

**Review by JulianaSau — APPROVED** [review]

File: packages/ui/src/message-pane/MessagePane.jsx (hunk 3 above)
Comment (JulianaSau) [inline packages/ui/src/message-pane/MessagePane.jsx:16]: Remove this console.log

**Review by JulianaSau — CHANGES_REQUESTED** [review]

**Comment by Johndiddles** [comment]
console.log has been removed.

**Review by JulianaSau — APPROVED** [review]

**Pull request merged**